### PR TITLE
Improve handling of resolved yet stuck method calls

### PR DESCRIPTION
**Pull request by kelloggm**
Makes the input in #298 compilable. The test case I've added on checks for part of the problem, but I verified the actual problem is fixed, too.

### CODE REVIEW

File: src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java [hunk 1]
```diff
@@ -1123,6 +1123,7 @@ public Visitable visit(MethodCallExpr method, Void p) {
       updateSyntheticClassForSuperCall(method);
       return super.visit(method, p);
     }
+    String methodName = method.getNameAsString();
```
Comment (kelloggm) [inline src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java:1126]: This change is unrelated, but it makes life simpler for the nullness checker

**Review by LoiNguyenCS — CHANGES_REQUESTED** [review]
Professor, I have a few questions regarding this PR.

File: src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java [hunk 2]
```diff
@@ -546,7 +546,28 @@ public Visitable visit(MethodCallExpr call, Void p) {
         // leading to a RuntimeException.
         // Note: this preservation is safe because we are not having an UnsolvedSymbolException.
         // Only unsolved symbols can make the output failed to compile.
-        resolvedYetStuckMethodCall.add(this.classFQName + "." + call.getNameAsString());
+        if (call.hasScope()) {
+          Expression scope = call.getScope().orElseThrow();
+          String scopeAsString = scope.toString();
+          if (scopeAsString.equals("this") || scopeAsString.equals("super")) {
```
[the remark this answers is not in the document]
Comment (kelloggm) [reply src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java:562]: It should be the FQN of either the superclass or an implemented interface, I think. TargetMethodFinderVisitor doesn't actually track that information, though, and without a test case for this situation I'm disinclined to add another data structure to keep track of it. I'll add a comment to the code here with a TODO, instead, and open an issue.

**Review by LoiNguyenCS — APPROVED** [review]
LGTM.

**Pull request merged**